### PR TITLE
Get the host machine ID when running on Windows

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -6980,6 +6986,31 @@ fu_engine_load_local_metadata_watches(FuEngine *self, GError **error)
...
+	LSTATUS rc;
+
+	rc = RegGetValue(hkey, subkey, value, RRF_RT_REG_SZ, NULL, (PVOID)&buf, &bufsz);
+	if (rc != ERROR_SUCCESS) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_INVAL,
+			    "Failed to get registry string %s [0x%lX]",
+			    subkey,
+			    (unsigned long)rc);
```
Comment (superm1) [inline src/fu-engine.c:7007]: Can you use GetLastError?

**Review by superm1 — COMMENTED** [review]
Have you tried on real Windows? I'm curious if this works without admin rights. Everything else so far does.

File: src/fu-engine.c (hunk 1 above)
Comment (hughsie) [reply src/fu-engine.c:7007]: I don't think it's the same thing, no?

**Comment by hughsie** [comment]
> Have you tried on real Windows

Nope, will do now. I've been testing in WINE.

**Comment by hughsie** [comment]
>  I'm curious if this works without admin rights

Seems to work in QEMU with win10.

**Review by superm1 — APPROVED** [review]

**Pull request merged**